### DISCUSSION TITLE
Knit to html R Studio memakan waktu lama (capstone ML)

### DISCUSSION THREAD

**Discussion by angeliainge** [teamalgoritma/community#335] [Q&A] [answered]
Dear tim Algoritma,

Saya ingin bertanya untuk proses knitting to html di R Studio saya memakan waktu yang sangat lama, sampai saat ini sudah stuck 5 jam di chunk ke 9 (dari 51 chunks).

Mohon panduannya apakah ada solusi?
so far yang sudah saya coba adalah:
1. set eval = FALSE, dengan hasil codingan saya saja tanpa hasil yang di run. Berikut saya lampirkan screenshot chunk ke 9 dan hasil knit dgn eval = false untuk referensi.
2. restart r
3. restart laptop
4. clear knitr cache
5. set chunk ke 9 dgn # agar tidak di run

saat knit tetap saja stuck kembali di chunk ke 9 lagi..
<img width="651" alt="Screen Shot 2022-09-23 at 5 25 29 AM" src="https://user-images.githubusercontent.com/107099548/191862083-87d550ca-81c3-4ed5-aece-f9a607ca93fc.png">

<img width="848" alt="Screen Shot 2022-09-23 at 4 56 48 AM" src="https://user-images.githubusercontent.com/107099548/191858677-2c7ec8be-a596-4435-b11c-ba44922d6491.png">

**Comment by NabiilahArdini** [comment, marked as answer]
Dear @angeliainge, proses yang lama ini dapat disebabkan oleh pemrosesan data yang juga dijalankan ketika knitting. Saran dari kami:
1. Hindari melakukan print data yang banyak (gunakan head untuk data tabular atau hanya print beberapa contoh saja).
2. Untuk mencegah proses code berjalan juga ketika knitting, Anda bisa menyimpan objek hasil processing ke dalam .RDS, menjadikan comment/`eval=FALSE` untuk code preprocessingnya, kemudian meload .RDS nya saja untuk proses2 selanjutnya.

Demikian yang dapat saya sarankan. Semoga dapat bermanfaat.

**Reply by angeliainge** [reply]
noted, thank u :)
